app/api/entities/entities.js

```diff
@@ -1,18 +1,34 @@
 import { generateNamesAndIds } from 'api/templates/utils';
 import ID from 'shared/uniqueID';
-import date from 'api/utils/date.js';
+import { templateTypes } from 'shared/templateTypes';
+import date from 'api/utils/date';
 import relationships from 'api/relationships/relationships';
 import createError from 'api/utils/Error';
 import search from 'api/search/search';
 import templates from 'api/templates/templates';
 import path from 'path';
 import PDF from 'api/upload/PDF';
 import paths from 'api/config/paths';
+import { wrapValidation } from 'api/utils/wrapValidation';
 
-import { deleteFiles } from '../utils/files.js';
+import { deleteFiles } from '../utils/files';
 import model from './entitiesModel';
+import validator from './entitiesValidator';
```

Comment:
does it happen in the rest of Uwazi that model.save() is called directly? I.e. should validation also happen as part of the model?

Reply:
Since the `entities.save()` is the central means by with entities are saved across Uwazi, I think it should handle validation to ensure that entities that are stored in the db are always valid regardless of whether they were created via the API, through a file import or through some other process in Uwazi.